x-pack/plugins/grokdebugger/public/plugin.js

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { first } from 'rxjs/operators';
+import { registerFeature } from './register_feature';
+import { PLUGIN } from '../common/constants';
+
+const inactiveLicenseMessage = i18n.translate('xpack.grokDebugger.clientInactiveLicenseError', {
+  defaultMessage: 'The Grok Debugger tool requires an active license.',
+});
+
+export class Plugin {
+  setup(coreSetup, plugins) {
+    registerFeature(plugins.home);
+
+    this.licensing = plugins.licensing;
+
+    this.licensing.license$.pipe(first()).subscribe(license => {
+      plugins.devTools.register({
```

Comment:
I would recommend keeping all registration synchronous in the plugin's setup function. Otherwise this could result in a race condition where actions considered setup phase occur outside of setup phase.

One approach here would be moving `this.licensing.license$.pipe` to inside of the `async mount` function. (see `x-pack/plugins/searchprofiler/public/plugin.ts` for an example).

Reply:
Yep, good point 👍 I'll fix this.